src/Swift.Runtime/src/Metadata/TypeMetadata.cs

```diff
@@ -171,6 +171,26 @@ public TypeMetadataKind Kind {
         }
     }
 
+    /// <summary>
+    /// Returns a pointer to the value witness table for the given type
+    /// </summary>
+    public unsafe ValueWitnessTable *ValueWitnessTable => (ValueWitnessTable*)(*((IntPtr*)handle - 1));
```

Comment:
Maybe this should throw on `handle == IntPtr.Zero`?

The main annoyance I see is that if you inspect/expand `TypeMetadata.Zero` in a debugger, the process will most likely crash, because it's accessing invalid memory.

Throwing a managed exception instead would be much better, which the debugger knows how to handle.